src/VolumeRenderSettings.ts

```diff
@@ -127,10 +127,10 @@ export class VolumeRenderSettings {
   }
 
   public resizeWithVolume(volume: Volume): void {
-    this.zSlice = Math.floor(volume.z / 2);
-    this.specular = new Array(volume.num_channels).fill([0, 0, 0]);
-    this.emissive = new Array(volume.num_channels).fill([0, 0, 0]);
-    this.glossiness = new Array(volume.num_channels).fill(0);
+    this.zSlice = Math.floor(volume.imageInfo.regionSize.z / 2);
+    this.specular = new Array(volume.imageInfo.numChannels).fill([0, 0, 0]);
+    this.emissive = new Array(volume.imageInfo.numChannels).fill([0, 0, 0]);
+    this.glossiness = new Array(volume.imageInfo.numChannels).fill(0);
```

Comment:
Nit: it would be cool if we could declare volume.imageInfo as read-only somehow, just to protect these deep accesses.  When I see these it makes me want to put wrapper functions so you write volume.getNumChannels()

Reply:
Adding `readonly` before all property values or wrapping the whole type in `Readonly` (which just maps lower-case `readonly` over the whole type) accomplishes this. I did the latter and fixed the couple direct modifications it revealed. The property doesn't make it all the way in to individual vectors: it's still possible to modify individual x/y/z components or call mutating methods. Some more clever type mapping could at least protect direct component modification though, if we want that.

Reply:
The C++ programmer in me always feels nervous when you have public property accesses 3+ levels deep.  🤷